src/js/messaging/containers/Compose.jsx

```diff
@@ -100,6 +105,22 @@ class Compose extends React.Component {
     this.props.deleteComposeMessage();
   }
 
+  handleAttachmentsChange(domEvent) {
+    const input = domEvent.target;
+    if (window.File && window.FileList) {
+      const files = Array.from(input.files);
+      this.props.setAttachments(files);
+    }
+  }
+
+  handleAttachmentDelete(domEvent) {
+    const attachment = JSON.parse(domEvent.currentTarget.dataset.args).attachment;
+    // Remove the requested attachment
+    this.props.message.attachments.splice(attachment, 1);
+    // Reset the attachments array
+    this.props.setAttachments(this.props.message.attachments);
```

Comment:
Removing an attachment should probably be its own action. Modifying a component's props is considered an anti-pattern.